pkg/controller/manager.go

```diff
@@ -161,8 +163,8 @@ func (m *Manager) removeAndReturnController(name string) (*managedController, er
 		return nil, fmt.Errorf("empty controller map")
 	}
 
-	oldCtrl, ok := m.controllers[name]
-	if !ok {
+	oldCtrl := m.lookupLocked(name)
```

Comment:
From concurrency point of view, there is no different from the delete line 164, 165.
Is this the intention ?

Reply:
Yes, there should not be any behavioral difference.